models/connections.go

```diff
@@ -48,3 +102,153 @@ func (c Connection) String() string {
 func (c Connection) AsMap(removeFields ...string) map[string]any {
 	return asMap(c, removeFields...)
 }
+
+// AsGoGetterURL returns the connection as a url that's supported by https://github.com/hashicorp/go-getter
+// Connection details are added to the url as query params
+func (c Connection) AsGoGetterURL() (string, error) {
+	parsedURL, err := url.Parse(c.URL)
+	if err != nil {
+		return "", err
+	}
+
+	var output string
+	switch strings.ReplaceAll(strings.ToLower(c.Type), " ", "_") {
+	case ConnectionTypeHTTP:
+		if c.Username != "" || c.Password != "" {
+			parsedURL.User = url.UserPassword(c.Username, c.Password)
+		}
+
+		output = parsedURL.String()
+
+	case ConnectionTypeGit:
+		q := parsedURL.Query()
+
+		if c.Certificate != "" {
+			q.Set("sshkey", base64.URLEncoding.EncodeToString([]byte(c.Certificate)))
+		}
+
+		if v, ok := c.Properties["ref"]; ok {
+			q.Set("ref", v)
+		}
+
+		if v, ok := c.Properties["depth"]; ok {
+			q.Set("depth", v)
+		}
+
+		parsedURL.RawQuery = q.Encode()
+		output = parsedURL.String()
+
+	case ConnectionTypeAWS:
+		q := parsedURL.Query()
+		q.Set("aws_access_key_id", c.Username)
+		q.Set("aws_access_key_secret", c.Password)
+
+		if v, ok := c.Properties["profile"]; ok {
+			q.Set("aws_profile", v)
+		}
+
+		if v, ok := c.Properties["region"]; ok {
+			q.Set("region", v)
+		}
+
+		// For S3
+		if v, ok := c.Properties["version"]; ok {
+			q.Set("version", v)
+		}
+
+		parsedURL.RawQuery = q.Encode()
+		output = parsedURL.String()
+	}
+
+	return output, nil
+}
+
+// AsEnv generates environment variables and a configuration file content based on the connection type.
+func (c Connection) AsEnv(ctx context.Context) EnvPrep {
+	var envPrep = EnvPrep{
+		Files: make(map[string]bytes.Buffer),
+	}
+
+	switch strings.ReplaceAll(strings.ToLower(c.Type), " ", "_") {
+	case ConnectionTypeAWS:
+		envPrep.Env = append(envPrep.Env, fmt.Sprintf("AWS_ACCESS_KEY_ID=%s", c.Username))
+		envPrep.Env = append(envPrep.Env, fmt.Sprintf("AWS_SECRET_ACCESS_KEY=%s", c.Password))
+
+		// credentialFilePath :="$HOME/.aws/credentials"
+		credentialFilePath := filepath.Join(".creds", "aws", fmt.Sprintf("cred-%d", rand.Intn(100000000)))
```

Comment:
@moshloop using `fs/FS` wasn't possible because it seems to be read-only. Can't create new files or write to one.

- https://github.com/golang/go/issues/45757